File: other-programming-languages/Kotlin/2024/2024-001/part-001/Main.kt
import java.io.File
import kotlin.math.abs

fun main() {
    // Define the file path
    val filePath = "input.txt"

    // Read the file into two lists
    val (leftList, rightList) = readInputFile(filePath)

    // Sort both lists
    val sortedLeft = leftList.sorted()
    val sortedRight = rightList.sorted()

    // Calculate the total distance
    val totalDistance = sortedLeft.zip(sortedRight).sumOf { (left, right) -> abs(left - right) }

    // Print the total distance
    println("Total Distance: $totalDistance")
}

// Function to read the input file and extract two lists
fun readInputFile(filePath: String): Pair<List<Int>, List<Int>> {
    val leftList = mutableListOf<Int>()
    val rightList = mutableListOf<Int>()

    File(filePath).useLines { lines ->
        lines.forEach { line ->
            val parts = line.split("\\s+".toRegex())
            if (parts.size == 2) {
                leftList.add(parts[0].toInt())
                rightList.add(parts[1].toInt())
            }
        }
    }

    return Pair(leftList, rightList)
}

File: other-programming-languages/Perl/2024/2024-001/part-001/main.pl
use strict;
use warnings;

# Define the input file path
my $file_path = "input.txt";

# Initialize arrays for the two lists
my (@left_list, @right_list);

# Read the file and extract the two columns
open my $fh, '<', $file_path or die "Could not open file '$file_path': $!";
while (my $line = <$fh>) {
    chomp $line;
    my @parts = split /\s+/, $line;
    if (@parts == 2) {
        push @left_list,  $parts[0];
        push @right_list, $parts[1];
    }
}
close $fh;

# Sort both lists
@left_list  = sort { $a <=> $b } @left_list;
@right_list = sort { $a <=> $b } @right_list;

# Calculate the total distance
my $total_distance = 0;
for my $i (0 .. $#left_list) {
    $total_distance += abs($left_list[$i] - $right_list[$i]);
}

# Print the result
print "Total Distance: $total_distance\n";
